ticker/ticker.go

```diff
@@ -732,27 +732,27 @@ func (t *Ticker) parsePacket(b []byte) (Tick, error) {
 		tick.Timestamp = kiteconnect.Time{time.Unix(int64(binary.BigEndian.Uint32(b[60:64])), 0)}
 		tick.NetChange = lastPrice - closePrice
 
-		// Depth.
-		var i, j int
-		for i = 64; i < 124; i += 12 {
-			tick.Depth.Buy[j] = DepthItem{
-				Quantity: binary.BigEndian.Uint32(b[i : i+4]),
-				Price:    t.convertPrice(seg, float64(binary.BigEndian.Uint32(b[i+4:i+8]))),
-				Orders:   uint32(binary.BigEndian.Uint16(b[i+8 : i+10])),
-			}
+		// Depth Information.
+		var (
+			buyPos  = 64
+			sellPos = 124
+		)
 
-			j++
-		}
+		for i = 0; i < 5; i++ {
```

Comment:
Instead of hardcoding the number of entries(5) you can declare var like `var itemsCount = (sellPos - buyPos)/10` where 10 is the size of individual item which will never change. In this way if packet structure changes we just update `buyPos` and `sellPos`.

Reply:
@vividvilla updated it.